### PR TITLE
Fully-Compressed Suffix Trees.

### PR DESCRIPTION
**Pull request by fermeise**
Implementation of fully-compressed suffix trees by Russo et al.

### CODE REVIEW

**Comment by simongog** [comment]
Thanks for the code. It seems that I have to add the cst_fully class to the CST tests ;)

**Comment by simongog** [comment]
ok to test

**Comment by simongog** [comment]
ok to test

File: include/sdsl/cst_fully.hpp [hunk 1]
```diff
@@ -0,0 +1,788 @@
...
+            size_t leaf = m_csa.psi[v.first];
+            return node_type(leaf, leaf);
+        }
+
+        return lca(m_csa.psi[v.first], m_csa.psi[v.second]);
+    }
+
+//! Compute the Weiner link of node v and character c.
+        /*
+         * \param v A valid not of a cst_fully.
```
Comment (eloj) [inline include/sdsl/cst_fully.hpp]: s/not/node/

Comment (simongog) [inline include/sdsl/cst_fully.hpp]: Thanks!

**Pull request merged**